ClientApp/viking-vault/src/RegisterForm.tsx

```diff
@@ -0,0 +1,233 @@
+            firstName: "",
+            lastName: "",
+            pictureLink: "",
+            address: "",
+            cnp: "",
+            valid: "",
+            status: -1,
+            response: "",
+            openModal: false,
+            redirect: false
+        }
+    }
+
+    private validateEmail = () => {
+        fetch(url+"UniqueEmail", {
+            method: "POST",
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+              email: this.state.email,
+            })
+        },).then(response => response.json()).then(result => {
+            if(result === "unique") {
+                this.setState({
+                    valid: "valid"
+                })
+            }
+            else if(result === "not-unique") {
+                this.setState({
+                    valid: "invalid"
+                })
+            }
+            else {
+                this.setState({
+                    valid: "null"
+                })
+            }
+        });
+    }
+
+    private handleChange = (val: string, inpName: string) => {
+        this.setState({
+            [inpName]: val
+        })
+    }
+
+    private handleEmailChange = (e: any) => {
+        this.setState({
+            email: e.target.value
+        }, this.validateEmail);
+    }
+
+    private emailError = () => {
+        if(!emailRegEx.test(this.state.email)) {
+            return  {
+                msg: "Email format incorrect",
+                class: "alert alert-danger"
+            }
+        }
+        if(this.state.valid === "invalid") {
+            return  {
+                msg: "Email already exists",
+                class: "alert alert-danger"
+            }
+        }
+        if(this.state.valid === "valid") {
+            return  {
+                msg: "Valid",
+                class: "alert alert-success"
+            }
+        }
+        return  {
+            msg: "Server error. Try again later",
+            class: "alert alert-danger"
+        }
+    }
+
+    private handleCPasswordChange = (e: any) => {
+        this.setState({
+            cpassword: e.target.value
+        }, this.passwordMatch)
+    }
+
+    private passwordMatch = () => {
+        if(this.state.password === this.state.cpassword)
+            return  {
+                msg: "Passwords match",
+                class: "alert alert-success"
+            }
+        return  {
+            msg: "Password do not match",
+            class: "alert alert-danger"
+        }
+    }
+
+    private mandatoryFieldsCompletedCorrectly = () => {
+        let val = true;
+        if(this.emailError().msg !== "Valid") {
+            val = false;
+        }
+        if(this.passwordMatch().msg !== "Passwords match") {
+            val = false;
+        }
+        if(this.state.address === "" || this.state.cnp === "" || this.state.firstName === "" 
```

Comment:
You could refactor this using the Array method of .every(), like `["address", "cnp", "firstName", "lastName"].every(keyName => this.state[keyName] !== "")`